library/public/types.h

```diff
@@ -58,6 +71,56 @@ class F3D_EXPORT ratio_t
   double Value;
 };
 
+/**
+ * Describe a RGB color.
+ */
+class F3D_EXPORT color_t
+{
+public:
+  color_t() = default;
+  color_t(const std::vector<double>& vec)
+  {
+    if (vec.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided vector does not have the right size");
+    }
+    std::copy_n(vec.begin(), 3, this->RGB.begin());
+  }
+  color_t(double red, double green, double blue)
+  {
+    this->RGB[0] = red;
+    this->RGB[1] = green;
+    this->RGB[2] = blue;
+  }
+  color_t(const std::initializer_list<double>& list)
+  {
+    if (list.size() != 3)
+    {
+      throw f3d::type_construction_exception("Provided list does not have the right size");
+    }
+    std::copy_n(list.begin(), 3, this->RGB.begin());
+  }
```

Comment:
```suggestion
  template<std::size_t N>
  color_t(const double (&list)[N])
  {
    static_assert(N == 3, "color_t needs exactly 3 elements to initialize");
    std::copy_n(list, 3, this->RGB.begin());
  }
```
Or something idk, I was just going to ask "can this be compile time?" but I ended copy-pasting stuff I barely understand instead https://cplusplus.github.io/CWG/issues/1591.html :shrug: